compiler/circle-mpqsolver/src/core/ErrorMetric.cpp

```diff
@@ -44,22 +44,28 @@ float MAEMetric::compute(const WholeOutput &first, const WholeOutput &second) co
       assert(first_elementary.size() == second_elementary.size());
       size_t cur_size = first_elementary.size() / loco::size(loco::DataType::FLOAT32);
 
+      double output_error = 0.; // mean error oevr current output
+
       const float *first_floats = reinterpret_cast<const float *>(first_elementary.data());
       const float *second_floats = reinterpret_cast<const float *>(second_elementary.data());
       for (size_t index = 0; index < cur_size; index++)
       {
-        float ref_value = *(first_floats + index);
-        float cur_value = *(second_floats + index);
-        error += std::fabs(ref_value - cur_value);
+        double ref_value = static_cast<double>(*(first_floats + index));
+        double cur_value = static_cast<double>(*(second_floats + index));
+        output_error += std::fabs(ref_value - cur_value);
+      }
+      if (cur_size != 0)
+      {
+        output_errors += (output_error / cur_size);
+        num_output_errors += 1;
       }
-      output_size += cur_size;
     }
   }
 
-  if (output_size == 0)
+  if (num_output_errors == 0)
   {
     throw std::runtime_error("nothing to compare");
   }
 
-  return error / output_size;
+  return static_cast<double>(output_errors / num_output_errors);
```

Comment:
Please update function signature too.

Reply:
If you intended to return `float`, then this line may need to be udpated (`static_cast<float>`).

Reply:
Ahhh. Sorry. Typo!

Reply:
fixed